src/main/java/org/acra/ACRAConfiguration.java

```diff
@@ -320,6 +321,12 @@ public ACRAConfiguration setResDialogNegativeButtonText(int resId) {
         return this;
     }
 
+    @SuppressWarnings( "unused" )
+    public ACRAConfiguration setReportDialogClass(Class<? extends CrashReportDialog> reportDialogClass) {

```

Comment:
There are places where isinstanceof is used which would need to be changed
to support an arbitrary activity being passed. I don't think it's really
worth it, since one can just override onCreate if they want a completely
new activity.

In the gist I'm going to switch on or off auto- report. I can show a more
complete example in a few days when I'm back home if you like?
On 16/01/2015 8:05 pm, "William Ferguson" <notifications@github.com> wrote:

> In src/main/java/org/acra/ACRAConfiguration.java
> <https://github.com/ACRA/acra/pull/218#discussion-diff-23074599>:
>
> > @@ -320,6 +321,12 @@ public ACRAConfiguration setResDialogNegativeButtonText(int resId) {
> >          return this;
> >      }
> >
> > +    @SuppressWarnings( "unused" )
> > +    public ACRAConfiguration setReportDialogClass(Class<? extends CrashReportDialog> reportDialogClass) {
>
> Yeah. That's where I have doubts. I like the idea, don't get me wrong. But
> this will open a whole can of maintenance worms.2 out of 3 will implement
> it backwards and I'll be fielding the support fall out. ᐧ
> … <#14af26b9cdde56cb_>
> On Fri, Jan 16, 2015 at 9:00 PM, npombourcq <notifications@github.com>
> wrote: In src/main/java/org/acra/ACRAConfiguration.java <
> https://github.com/ACRA/acra/pull/218#discussion-diff-23074370>: > @@
> -320,6 +321,12 @@ public ACRAConfiguration
> setResDialogNegativeButtonText(int resId) { > return this; > } > > +
> @SuppressWarnings <https://github.com/SuppressWarnings>( "unused" ) > +
> public ACRAConfiguration setReportDialogClass(Class<? extends
> CrashReportDialog> reportDialogClass) { At that point it's really up to the
> custom implementation to decide what they want to do. They can extend to
> get most of the base behavior as is, or duplicate from CrashReportDialog if
> they have somewhat different logic. — Reply to this email directly or view
> it on GitHub <https://github.com/ACRA/acra/pull/218/files#r23074370>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ACRA/acra/pull/218/files#r23074599>.
>